modules/apps/document-library-opener/document-library-opener-service/src/main/java/com/liferay/document/library/opener/internal/upload/UniqueFileEntryTitleProviderImpl.java

```diff
@@ -74,12 +85,42 @@ private boolean _exists(long groupId, long folderId, String fileName) {
 		return false;
 	}
 
-	private String _provide(long groupId, long folderId, String fileName)
+	private boolean _existsTitle(long groupId, long folderId, String title) {
+		try {
+			_dlAppLocalService.getFileEntry(groupId, folderId, title);
+
+			return true;
+		}
+		catch (NoSuchFileEntryException noSuchFileEntryException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(noSuchFileEntryException, noSuchFileEntryException);
+			}
+		}
+		catch (PortalException portalException) {
+			throw new SystemException(portalException);
+		}
+
+		return false;
+	}
+
+	private String _provide(long groupId, long folderId, String title)
+		throws PortalException {
+
+		return _uniqueFileNameProvider.provide(
```

Comment:
Hi @AliciaGarciaGarcia, should we check here if the fileName exists too?

Reply:
I see we don't have the extension, so it is not possible.

Reply:
In this method, without the extension, we cannot check it.
There is no restriction that the filename on one file should not be the title of another (extrange, yes, but not illegal)
We can deprecate the method